api/src/services/project-service.ts

```diff
@@ -785,56 +785,56 @@ export class ProjectService extends DBService {
     await Promise.all([...insertActivityPromises]);
   }
 
+  /**
+   * Compares incoming project funding data against the existing funding data, if any, and determines which need to be
+   * deleted, added, or updated.
+   *
+   * @param {number} projectId
+   * @param {IUpdateProject} entities
+   * @return {*}  {Promise<void>}
+   * @memberof ProjectService
+   */
   async updateFundingData(projectId: number, entities: IUpdateProject): Promise<void> {
-    const putFundingData = entities?.funding && new PutFundingData(entities.funding);
+    const projectRepository = new ProjectRepository(this.connection);
 
-    const fundingIds = putFundingData?.fundingSources.map((data) => {
-      return data.id;
-    });
+    const putFundingData = entities?.funding && new PutFundingData(entities.funding);
+    if (!putFundingData) {
+      throw new HTTP400('Failed to create funding data object');
+    }
 
-    const surveyFundingSourceDeleteStatement = queries.survey.deleteSurveyFundingSourceConnectionToProjectSQL(
-      fundingIds
-    );
+    const existingProjectFundingSources = await projectRepository.getProjectFundingSourceIds(projectId);
 
-    if (surveyFundingSourceDeleteStatement) {
-      const surveyFundingSourceDeleteResult = await this.connection.query(
-        surveyFundingSourceDeleteStatement.text,
-        surveyFundingSourceDeleteStatement.values
+    const existingFundingSourcesToDelete = existingProjectFundingSources.filter((existingFunding) => {
```

Comment:
it would be nice to have a comment explaining what this is trying to achieve, hard to tell at first glance